src/platform_impl/linux/wayland/pointer.rs

```diff
@@ -101,13 +110,16 @@ pub fn implement_pointer(
                         surface_y,
                         ..
                     } => {
-                        if let Some(wid) = mouse_focus {
+                        if let (Some(wid), Some(surface)) = (mouse_focus, current_surface.as_ref())
```

Comment:
This could be make simpler by putting both `surface` and `wid` in the same option as a tuple.

Or even by storing only `surface` and recreating the `wid` on the fly using `make_wid` whenever needed, as it is known to be a `surface` corresponding to a winit window thanks to the initial check of `find_wid` in `PtrEvent::Enter`.

Reply:
>Or even by storing only surface and recreating the wid on the fly using make_wid whenever needed, as it is known to be a surface corresponding to a winit window thanks to the initial check of find_wid in PtrEvent::Enter.

I like this idea.